specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/LocationCapabilities.json

```diff
@@ -579,6 +580,119 @@
         }
       }
     },
+    "JobAgentEditionCapability": {
+      "description": "The job agent edition capability.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The job agent edition name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "supportedServiceLevelObjectives": {
+          "description": "The list of supported service level objectives for the edition.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/JobAgentServiceLevelObjectiveCapability"
+          },
+          "readOnly": true,
+          "x-ms-identifiers": []
+        },
+        "status": {
+          "description": "The status of the capability.",
+          "enum": [
+            "Visible",
+            "Available",
+            "Default",
+            "Disabled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "CapabilityStatus",
+            "modelAsString": false
+          }
+        },
+        "reason": {
+          "description": "The reason for the capability not being available.",
+          "type": "string"
+        }
+      }
+    },
+    "JobAgentServiceLevelObjectiveCapability": {
+      "description": "The job agent service level objective capability.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The service objective name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "sku": {
+          "$ref": "../../../common/v1/types.json#/definitions/Sku",
+          "description": "The sku.",
+          "readOnly": true
+        },
+        "status": {
+          "description": "The status of the capability.",
+          "enum": [
+            "Visible",
+            "Available",
+            "Default",
+            "Disabled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "CapabilityStatus",
+            "modelAsString": false
+          }
+        },
+        "reason": {
+          "description": "The reason for the capability not being available.",
+          "type": "string"
+        }
+      }
+    },
+    "JobAgentVersionCapability": {
+      "description": "The job agent version capability.",
+      "type": "object",
+      "properties": {
+        "name": {
```

Comment:
>name [](http://example.com/codeflow?start=9&length=4)

Don't use the ARM top-level property names (name, type, id, systemData) for properties within the property bag. Same comment applies for 2 other "name" properties.

Reply:
This is how ElasticPoolEditionCapability, InstancePoolVcoresCapability, and other objects in this file are structured. This is the format they have had for several API versions now and we are seeking consistency with them.

Reply:
There are some apis or *parts* of apis, that don't follow this guidance, but in general you want to avoid it in properties, because if you flatten away the properties in the SDK code generation, which is good for usability, it becomes a duplicate 'name' conflict. [Not to mention a point of potentially confusing people as to why there are so many names.]

Reply:
I think in this context its acceptable usage.